### PR TITLE
Fix: Add compiler flags for tvOS

### PR DESCRIPTION
**Pull request by hufkens**
At the moment the lib doesn't compile on tvOS. 
I added the necessary compiler flags.

The code was tested in our tvOS/iOS apps.

### CODE REVIEW

**Comment by WoLewicki** [comment]
Could you double-check if all the code fragments with the unavailable APIs are covered by PR? If so, I would love to merge it.

**Comment by keshavkaul** [comment]
I suggest using `TARGET_OS_IOS` compiler flag as a positive check for including the compatible API.

This way, the check becomes intentional for supporting that target platform only and is more maintainable

File: ios/RNSScreenStackHeaderConfig.m [hunk 1]
```diff
@@ -375,6 +380,7 @@ + (void)updateViewController:(UIViewController *)vc withConfig:(RNSScreenStackHe
     }
   } else {
     prevItem.backBarButtonItem = nil;
+
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:383]:
```suggestion
```

File: ios/RNSScreenStackHeaderConfig.m [hunk 2]
```diff
@@ -357,7 +361,8 @@ + (void)updateViewController:(UIViewController *)vc withConfig:(RNSScreenStackHe
   }
 
   navitem.title = config.title;
-  if (config.backTitle != nil || config.backTitleFontFamily || config.backTitleFontSize) {
+#if (!TARGET_OS_TV)
+if (config.backTitle != nil || config.backTitleFontFamily || config.backTitleFontSize) {
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:365]:
```suggestion
  if (config.backTitle != nil || config.backTitleFontFamily || config.backTitleFontSize) {
```

File: ios/RNSScreenStackHeaderConfig.m [hunk 3]
```diff
@@ -383,6 +389,8 @@ + (void)updateViewController:(UIViewController *)vc withConfig:(RNSScreenStackHe
     }
     navitem.largeTitleDisplayMode = config.largeTitle ? UINavigationItemLargeTitleDisplayModeAlways : UINavigationItemLargeTitleDisplayModeNever;
   }
+    
+#endif
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:393]:
```suggestion
#endif

```

**Comment by WoLewicki** [comment]
Please apply these changes and I think it will be ready to be merged.

**Comment by WoLewicki** [comment]
@hufkens will you be able to apply these changes in the next few days? I was planning on making a release probably on Monday and it would be very nice to have this merged by then.

**Comment by olafurnsig** [comment]
Theres also on line 536 

` [blurEffects addEntriesFromDictionary:@{
      @"systemUltraThinMaterial": @(UIBlurEffectStyleSystemUltraThinMaterial),
      @"systemThinMaterial": @(UIBlurEffectStyleSystemThinMaterial),
      @"systemMaterial": @(UIBlurEffectStyleSystemMaterial),
      @"systemThickMaterial": @(UIBlurEffectStyleSystemThickMaterial),
      @"systemChromeMaterial": @(UIBlurEffectStyleSystemChromeMaterial),
      @"systemUltraThinMaterialLight": @(UIBlurEffectStyleSystemUltraThinMaterialLight),
      @"systemThinMaterialLight": @(UIBlurEffectStyleSystemThinMaterialLight),
      @"systemMaterialLight": @(UIBlurEffectStyleSystemMaterialLight),
      @"systemThickMaterialLight": @(UIBlurEffectStyleSystemThickMaterialLight),
      @"systemChromeMaterialLight": @(UIBlurEffectStyleSystemChromeMaterialLight),
      @"systemUltraThinMaterialDark": @(UIBlurEffectStyleSystemUltraThinMaterialDark),
      @"systemThinMaterialDark": @(UIBlurEffectStyleSystemThinMaterialDark),
      @"systemMaterialDark": @(UIBlurEffectStyleSystemMaterialDark),
      @"systemThickMaterialDark": @(UIBlurEffectStyleSystemThickMaterialDark),
      @"systemChromeMaterialDark": @(UIBlurEffectStyleSystemChromeMaterialDark),
    }];`

This is not supported by tvOS.

**Comment by olafurnsig** [comment]
Aaaand, this merge would be very much welcome :)

**Comment by WoLewicki** [comment]
@olafurnsig can you place these review comments in #598 since it is a followup of this PR? I will close this PR to stop the confusion. If the author applies all the changes requested in #598 before that one does, I can merge this PR then too.

**Comment by olafurnsig** [comment]
@WoLewicki he has added macros that fix the above issues in #598

**Pull request closed without merging**